app/src/androidTest/java/ch/hikemate/app/ui/auth/CreateAccountScreenTest.kt

```diff
@@ -0,0 +1,113 @@
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.verify
+
+class CreateAccountScreenTest : TestCase() {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var navigationActions: NavigationActions
+  private lateinit var authRepository: AuthRepository
+  private lateinit var authViewModel: AuthViewModel
+
+  private val TEST_NAME = "Test Name"
+  private val TEST_EMAIL = "test@example.com"
+  private val TEST_PASSWORD = "password"
+  private val TEST_DIFFERENT_PASSWORD = "different_password"
+
+  @Before
+  fun setUp() {
+    navigationActions = mock(NavigationActions::class.java)
+    authRepository = mock(AuthRepository::class.java)
+    authViewModel = AuthViewModel(authRepository)
+
+    composeTestRule.setContent {
+      CreateAccountScreen(navigationActions = navigationActions, authViewModel = authViewModel)
+    }
+  }
+
+  @Test
+  fun everythingIsOnScreen() {
+    composeTestRule.onNodeWithTag(BackButton.BACK_BUTTON_TEST_TAG).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(CreateAccountScreen.TEST_TAG_TITLE).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(CreateAccountScreen.TEST_TAG_NAME_INPUT).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(CreateAccountScreen.TEST_TAG_EMAIL_INPUT).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(CreateAccountScreen.TEST_TAG_PASSWORD_INPUT).assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_CONFIRM_PASSWORD_INPUT)
+        .assertIsDisplayed()
+    composeTestRule.onNodeWithTag(CreateAccountScreen.TEST_TAG_SIGN_UP_BUTTON).assertIsDisplayed()
+  }
+
+  @Test
+  fun canWriteToAllFields() {
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_NAME_INPUT)
+        .performTextInput(TEST_NAME)
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_EMAIL_INPUT)
+        .performTextInput(TEST_EMAIL)
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_PASSWORD_INPUT)
+        .performTextInput(TEST_PASSWORD)
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_CONFIRM_PASSWORD_INPUT)
+        .performTextInput(TEST_PASSWORD)
+  }
+
+  @Test
+  fun clickOnCreateAccountButton() {
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_NAME_INPUT)
+        .performTextInput(TEST_NAME)
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_EMAIL_INPUT)
+        .performTextInput(TEST_EMAIL)
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_PASSWORD_INPUT)
+        .performTextInput(TEST_PASSWORD)
+    composeTestRule
+        .onNodeWithTag(CreateAccountScreen.TEST_TAG_CONFIRM_PASSWORD_INPUT)
+        .performTextInput(TEST_PASSWORD)
+    composeTestRule.onNodeWithTag(CreateAccountScreen.TEST_TAG_SIGN_UP_BUTTON).performClick()
+
+    verify(authRepository, times(1))
+        .createAccountWithEmailAndPassword(any(), any(), eq(TEST_EMAIL), eq(TEST_PASSWORD))
```

Comment:
Remark: Not your task, but that doesn't make sense to have callbacks in the beginning of the method.... They should be at the end of the parameters list. Just so that we know it and think to put it as a new task

Reply:
I fixed it anyway

Reply:
Okay 😂